src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -28,199 +30,72 @@
-                break;
-            }
-            if (std::errc::result_out_of_range == error) {
-                SPDLOG_ERROR("File contains an incomplete IR stream");
-                break;
-            }
-            throw ClpFfiJsException{
-                    clp::ErrorCode::ErrorCode_Corrupt,
-                    __FILENAME__,
-                    __LINE__,
-                    "Failed to deserialize: "s + error.category().name() + ":" + error.message()
+                "this is irv1; gg."
             };
         }
-        m_stream_reader_data_context.reset(nullptr);
-    }
-
-    return m_encoded_log_events.size();
-}
-
-auto StreamReader::decode_range(size_t begin_idx, size_t end_idx) const -> DecodedResultsTsType {
-    if (m_encoded_log_events.size() < end_idx || begin_idx >= end_idx) {
-        return DecodedResultsTsType(emscripten::val::null());
-    }
-
-    std::span const log_events_span{
-            m_encoded_log_events.begin()
-                    + static_cast<decltype(m_encoded_log_events)::difference_type>(begin_idx),
-            m_encoded_log_events.begin()
-                    + static_cast<decltype(m_encoded_log_events)::difference_type>(end_idx)
-    };
-    std::string message;
-    constexpr size_t cDefaultReservedMessageLength{512};
-    message.reserve(cDefaultReservedMessageLength);
-    size_t log_num{begin_idx + 1};
-    auto const results{emscripten::val::array()};
-    for (auto const& log_event : log_events_span) {
-        message.clear();
-
-        auto const parsed{log_event.get_message().decode_and_unparse()};
-        if (false == parsed.has_value()) {
-            SPDLOG_ERROR("Failed to decode message.");
-            break;
-        }
-        message.append(parsed.value());
+        SPDLOG_INFO("The version is {}", version);
 
-        constexpr size_t cLogLevelPositionInMessages{1};
-        size_t log_level{cLogLevelNone};
-        // NOLINTNEXTLINE(readability-qualified-auto)
-        auto const log_level_name_it{std::find_if(
-                cLogLevelNames.begin() + cValidLogLevelsBeginIdx,
-                cLogLevelNames.end(),
-                [&](std::string_view level) {
-                    return message.substr(cLogLevelPositionInMessages).starts_with(level);
-                }
-        )};
-        if (log_level_name_it != cLogLevelNames.end()) {
-            log_level = std::distance(cLogLevelNames.begin(), log_level_name_it);
-        }
-
-        m_ts_pattern.insert_formatted_timestamp(log_event.get_timestamp(), message);
-
-        EM_ASM(
-                { Emval.toValue($0).push([UTF8ToString($1), $2, $3, $4]); },
-                results.as_handle(),
-                message.c_str(),
-                log_event.get_timestamp(),
-                log_level,
-                log_num
-        );
-        ++log_num;
+        return std::make_unique<KVPairIRStreamReader>(KVPairIRStreamReader::create(std::move(data_array)));
```

Comment:
This seems a bad practice since we have to recreate the data_buffer in KVPairIRStreamReader's factory function. Shall we make KVPairIRStreamReader's constructor public / friend so that we can
1. seek(0) in the zstd reader
2. create the deserializer from the zstd reader and then the StreamReaderDataContext
3. call KVPairIRStreamReader's constructor here.

so long we don't expose the constructor via Embind there should be minor concerns of misuses of the constructor (at the moment we are not exposing StreamReaderDataContext anyways). What do you think?